### PR TITLE
Remove unnecessary FHIR Json objects for de/serialization

### PR DESCRIPTION
**Pull request by jingtang10**
Also:
- Refactor tests with better structure and variable naming
- Improve README with better structure, more code examples on working with FHIR resources as well as minor formatting and grammar fixes
- Move version from build file to `gradle.properties`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -642,15 +704,8 @@ These tests are set up to run on JVM and as Android unit tests. To run them loca
...
-For a comprehensive understanding of publishing KMP libraries to Maven Central, see the
-[Kotlin Multiplatform Publishing Guide](https://kotlinlang.org/docs/multiplatform-publish-lib.html)
-and the
-[Maven Central Publishing Guide](https://central.sonatype.org/publish/publish-portal-guide/).
-
-> **Note:** The project has already been set up to be released to Maven using the
-> [`gradle-maven-publish-plugin`](https://github.com/vanniktech/gradle-maven-publish-plugin). The
-> following sections outline the additional setup required for a developer to publish to Maven Local
-> and Maven Central.
+To publish a new release, first update `mavenVersion` in `gradle.properties` to the new version.
```
Comment (Snipesy) [inline README.md:779]: With `mavenVersion` in gradle.properties you can tie the publish to a tag like this in `publish.yml`

```
  run: |
    TAG="${{ github.event.release.tag_name }}"
    VERSION="${TAG#v}"   # strips a leading "v" if present
    ./gradlew publish -PmavenVersion="$VERSION"
```

This gives a lot more flexibility (although getting the github action runners to build without OOM is a challenge on its own).

File: README.md [hunk 2]
```diff
@@ -625,6 +675,18 @@ executed:
    - Conversion to resource: Build a new FHIR resource object using `build()` function
    - Verification: The reconstructed object from the builder is equal to the original object.
 
+Additionally, there are targeted test suites for serialization edge cases:
```
Comment (Snipesy) [inline README.md]: I dont think new block adds much value to the readme.

**Review by Snipesy — CHANGES_REQUESTED** [review]

File: fhir-model/src/commonTest/kotlin/dev/ohs/fhir/model/test/SimpleSerializationTest.kt [hunk 3]
```diff
@@ -34,141 +33,94 @@ private val expectedPatientJson =
...
-
-    test("Serialized resource in R5 should have resourceType property") {
-      val patient = dev.ohs.fhir.model.r5.Patient(id = "patient-01")
-      FhirR5Json().encodeToString(patient).shouldBe(expectedPatientJson)
-    }
-
-    // Standalone path: `Patient.serializer()` resolves to `PatientSerializer` (the variant whose
-    // descriptor has `resourceType` at slot 0). The serializer writes the discriminator itself —
-    // no kotlinx-json polymorphic plumbing involved.
     val standaloneJson = Json { prettyPrint = true }
```
Comment (ellykits) [inline fhir-model/src/commonTest/kotlin/dev/ohs/fhir/model/test/SimpleSerializationTest.kt]: Nitpick: You can re-use the `Json` from `TestJson.kt` file here.

File: README.md [hunk 4]
```diff
@@ -29,45 +29,36 @@ Kotlin FHIR is a lean and fast implementation of the
...
+| Android applications and libraries | `android`     | `-android`      | ✅       |
+
+The library also supports the following
+[Kotlin/Native targets](https://kotlinlang.org/docs/native-target-support.html):
+
+| Gradle target     | Artifact suffix      | Tier | Support |
+|:------------------|:---------------------|:-----|:--------|
+| iosSimulatorArm64 | `-iossimulatorarm64` | 1    | ✅       |
+| iosArm64          | `-iosarm64`          | 1    | ✅       |
+| iosX64            | `-iosx64`            | 3    | ✅       |
```
Comment (ellykits) [inline README.md:62]: As a quick note, plans to deprecate the iosX64 target were temporarily paused, but support is expected to be removed in upcoming releases (likely Kotlin 2.4+) - [see changes to Apple target support](https://kotlinlang.org/docs/whatsnew23.html#changes-to-apple-target-support). Fortunately, our codebase already supports both iosSimulatorArm64 and iosArm64, so this shouldn't be a big deal.

Comment (jingtang10) [reply README.md:62]: yeah i think it's already tier 3 target now. we can probably strip it soon to be honest.

File: README.md (hunk 1 above)
Comment (jingtang10) [reply README.md:779]: does this mean we'll use whatever tag used when creating a github release as the version number? i'm personally more inclined for version numbers to be version-controlled in git. i think that reduces the chance of getting it wrong.

File: fhir-model/src/commonTest/kotlin/dev/ohs/fhir/model/test/SimpleSerializationTest.kt (hunk 3 above)
Comment (jingtang10) [reply fhir-model/src/commonTest/kotlin/dev/ohs/fhir/model/test/SimpleSerializationTest.kt]: also refactored this class

File: README.md (hunk 2 above)
Comment (jingtang10) [reply README.md]: removed

**Comment by jingtang10** [comment]
@Snipesy @ellykits please take another look. thanks

**Review by ellykits — APPROVED** [review]
Thanks for improving the documentation @jingtang10. It reads better 👍
